docs/contributing.rst

```diff
@@ -0,0 +1,267 @@
+To take effect, the changes have to be committed afterwards
+
+$ git commit -m "Updated submodules"
+
+Pull Request Guidelines
+-----------------------
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+2. If the pull request adds functionality, the docs should be updated. Put your new functionality into a function with a docstring.
+3. If checks do not pass, have a look at https://app.circleci.com/pipelines/github/pyfar/pyfar for more information.
+
+Function and Class Guidelines
+-----------------------------
+
+Functions and classes should
+
+* have a single clear purpose and a functionality limited to that purpose. Conditional parameters are fine in some cases but are an indicator that a function or class does not have a clear purpose. Conditional parameters are
+
+  - parameters that are obsolete if another parameter is provided
+  - parameters that are necessary only if another parameter is provided
+  - parameters that must have a specific value depending on other parameters
+
+* be split into multiple functions or classes if the functionality not well limited.
+* contain documentation for all input and output parameters.
+* contain examples in the documentation if they are non-trivial to use.
+* contain comments in the code that explain decisions and parts that are not trivial to read from the code. As a rule of thumb, too much comments are better than to little comments.
+* use clear names for all variables
+
+It is also a good idea to follow `the Zen of Python <https://peps.python.org/pep-0020/>`_
+
+Errors should be raised if
+
+* Audio objects do not have the correct type (e.g. a TimeData instance is passed but a Signal instance is required)
+* String input that specifies a function option has an invalid value (e.g. 'linea' was passed but 'linear' was required)
+* Invalid parameter combinations are used
+
+Warnings should be raised if
+
+* Results might be wrong or unexpected
+* Possibly bad parameter combinations are used
+
+
+Testing Guidelines
+-----------------------
+Pyfar uses test-driven development based on `three steps <https://martinfowler.com/bliki/TestDrivenDevelopment.html>`_ and `continuous integration <https://en.wikipedia.org/wiki/Continuous_integration>`_ to test and monitor the code.
+In the following, you'll find a guideline. Note: these instructions are not generally applicable outside of pyfar.
+
+- The main tool used for testing is `pytest <https://docs.pytest.org/en/stable/index.html>`_.
+- All tests are located in the *tests/* folder.
+- Make sure that all important parts of pyfar are covered by the tests. This can be checked using *coverage* (see below).
+- In case of pyfar, mainly **state verification** is applied in the tests. This means that the outcome of a function is compared to a desired value (``assert ...``). For more information, it is refered to `Martin Fowler's article <https://martinfowler.com/articles/mocksArentStubs.html.>`_.
```

Comment:
```suggestion
- In case of pyfar, mainly **state verification** is applied in the tests. This means that the outcome of a function is compared to a desired value (``assert ...``). For more information, it is refered to `Martin Fowler's article <https://martinfowler.com/articles/mocksArentStubs.html>`_.
```